spec/services/resource_compare_service_spec.rb

```diff
@@ -54,4 +55,112 @@
+      keywords: [],
+      contributors: [],
+      creators: []
+    }
+    expected_diff = {
+      doi: [{ action: :changed, from: "10.34770/pe9w-x904", to: "" }],
+      ark: [{ action: :changed, from: "ark:/88435/dsp01zc77st047", to: "" }],
+      description: [{ action: :changed,
+                      from: "All data is related to the Shakespeare and Company bookshop and lending library opened and operated by Sylvia Beach in Paris, 1919–1962.",
+                      to: "new description" }],
+      publication_year: [{ action: :changed, from: "2020", to: "" }],
+      version_number: [{ action: :changed, from: "1", to: "" }],
+      publisher: [{ action: :changed, from: "Princeton University", to: "" }],
+      resource_type: [{ action: :changed, from: "Dataset", to: "" }],
+      rights: [{ action: :changed, from: "Creative Commons Attribution 4.0 International", to: "" }],
+      titles: [{ action: :changed, from: "Shakespeare and Company Project Dataset: Lending Library Members, Books, Events ()", to: "new title ()" }],
+      collection_tags: [{ action: :changed, from: "", to: "fake" }],
+      creators: [{ action: :changed, from: "Kotin, Joshua | 1 | ", to: "" }]
+    }
+    resource1 = FactoryBot.create(:shakespeare_and_company_work).resource
+    keys = resource1.as_json.keys.sort
+    keys.each do |key|
+      # If a new key is added to the resource, but the ResourceCompareService misses it,
+      # we'll still test it here, and there should be a failure.
+      it "correctly compares #{key}" do
+        key_sym = key.to_sym
+        setter = "#{key}=".to_sym
+        resource2 = FactoryBot.create(:shakespeare_and_company_work).resource
+        resource2.send(setter, new_values[key_sym])
+        compare = described_class.new(resource1, resource2)
+        expect(compare.differences[key_sym]).to eq expected_diff[key_sym]
+      end
+    end
+  end
+
+  describe "generic comparisons" do
+    class MockResource < OpenStruct
+      # We want to confirm that the compare service still works with new unexpected attributes.
+      # This lets us treat a hash as if it were a resource.
+      def as_json
+        to_h
+      end
+    end
+
+    class MockComparable < OpenStruct
+      def compare_value
+        to_h.to_yaml.sub("---\n", "")
+      end
+    end
+
+    describe "value comparison" do
+      it "does not flag integer->string as change" do
+        compare = described_class.new(MockResource.new({ fake_year: 2000 }), MockResource.new({ fake_year: "2000" }))
+        differences = compare.differences[:fake_year]
+        expect(differences).to eq nil
+      end
+
+      it "does flag integer->string+1 as change" do
+        compare = described_class.new(MockResource.new({ fake_year: 2000 }), MockResource.new({ fake_year: "2001" }))
+        differences = compare.differences[:fake_year]
+        expect(differences).to eq [{ action: :changed, from: "2000", to: "2001" }]
+      end
+
+      it "does flag nil->string as change" do
```

Comment:
I believe this is string->nil
```suggestion
      it "does flag string->nil as change" do
```